### PR TITLE
[Redis][Part 2] Redis crossdock tests

### PR DESCRIPTION
**Pull request by AlexAPB**
This adds the redis transport to the existing oneway crossdock tests

### CODE REVIEW

File: internal/crossdock/server/oneway/server.go [hunk 1]
```diff
@@ -22,21 +22,33 @@ package oneway
...
+const redisAddr = "redis:6379"
+
 // Start starts the test server that clients will make requests to
 func Start() {
 	inbounds := []transport.Inbound{http.NewInbound(":8084")}
 
+	if redisIsAvailable() {
+		rds := redis.NewInbound(
+			redis.NewRedis5Client(redisAddr),
+			"yarpc/oneway", "yarpc/oneway/processing", time.Second)
```
Comment (willhug) [inline internal/crossdock/server/oneway/server.go]:
```
  "yarpc/oneway",
  "yarpc/oneway/processing",
  time.Second,
)

File: docker-compose.yml [hunk 2]
```diff
@@ -29,7 +30,7 @@ services:
             - AXIS_HTTPSERVER=go
             - AXIS_CLIENT_ONEWAY=go
             - AXIS_SERVER_ONEWAY=go
-            - AXIS_TRANSPORT_ONEWAY=http
+            - AXIS_TRANSPORT_ONEWAY=http,redis
```
Comment (breerly) [inline docker-compose.yml]: This is awesome.

File: docker-compose.yml [hunk 3]
```diff
@@ -5,6 +5,7 @@ services:
         image: crossdock/crossdock
         dns_search: .
         links:
+            - redis
```
Comment (breerly) [inline docker-compose.yml]: Lets move this to the last link, the primary containers are the test participants.

File: internal/crossdock/client/dispatcher/dispatcher.go [hunk 4]
```diff
@@ -94,6 +95,13 @@ func CreateOnewayDispatcher(t crossdock.T, handler raw.OnewayHandler) (yarpc.Dis
 				hostport.PeerIdentifier(fmt.Sprintf("%s:8084", server)),
 				http.NewTransport(),
 			))
+
```
Comment (breerly) [inline internal/crossdock/client/dispatcher/dispatcher.go]: Abhinav would say that this space should go.

File: internal/crossdock/client/dispatcher/dispatcher.go [hunk 5]
```diff
@@ -94,6 +95,13 @@ func CreateOnewayDispatcher(t crossdock.T, handler raw.OnewayHandler) (yarpc.Dis
...
+	case "redis":
+		outbound = redis.NewOnewayOutbound(
+			redis.NewRedis5Client("redis:6379"),
+			"yarpc/oneway",
+		)
+
```
Comment (breerly) [inline internal/crossdock/client/dispatcher/dispatcher.go]: And this one

File: internal/crossdock/server/oneway/server.go [hunk 6]
```diff
@@ -22,21 +22,36 @@ package oneway
...
 
+const redisAddr = "redis:6379"
+
 // Start starts the test server that clients will make requests to
 func Start() {
 	inbounds := []transport.Inbound{http.NewInbound(":8084")}
 
+	if redisIsAvailable() {
+		rds := redis.NewInbound(
+			redis.NewRedis5Client(redisAddr),
```
Comment (breerly) [inline internal/crossdock/server/oneway/server.go]: Suggest moving the addr inline. These tests are used by customers as docs, having the addr inline makes it easier to understand (I don't have to deref the const in my mind).

File: internal/crossdock/server/oneway/server.go [hunk 7]
```diff
@@ -63,3 +78,8 @@ func Stop() {
 		fmt.Println("failed to stop:", err.Error())
 	}
 }
+
+// redisIsAvailable checks to see if a redis server is available
+func redisIsAvailable() bool {
```
Comment (breerly) [inline internal/crossdock/server/oneway/server.go]: Suggest `redisIsExpectedRunning`.

**Pull request merged**